core/vm/analysis.go

```diff
@@ -41,21 +41,47 @@ func (d destinations) has(codehash common.Hash, code []byte, dest *big.Int) bool
 		m = jumpdests(code)
 		d[codehash] = m
 	}
-	return (m[udest/8] & (1 << (udest % 8))) != 0
+	return OpCode(code[udest]) == JUMPDEST && (m[udest/8]&(0x80>>(udest%8))) == 0
+	//	return (m[udest/8] & (1 << (udest % 8))) != 0
+}
+
+type bitvec struct {
+	m []byte
+}
+
+func (bits *bitvec) addone(pos uint64) {
+	bits.m[pos/8] |= 0x80 >> (pos % 8)
+}
+func (bits *bitvec) addOneByte(pos uint64) {
+	bits.m[pos/8] |= 0xFF >> (pos % 8)
+	bits.m[pos/8+1] |= ^(0xFF >> (pos % 8))
 }
```

Comment:
The naming could be better. I think`addone` can be `set` and `addOneByte` can be `set8` or something like it. It's confusing to say 'byte' because bitvec is []byte.